### PR TITLE
BUG: plotting grouped_hist with a single row frame #10214

### LINKED ISSUE #10214: hist raises AttributeError: 'AxesSubplot' object has no attribute 'ndim' with a one row DataFrame
Hello,

this code works fine:

    import pandas as pd
    import numpy as np
    import matplotlib.pyplot as plt
    bins = np.arange(80, 100 + 2, 1)
    df = pd.DataFrame({"Name": ["AAA", "BBB"], "ByCol": [1, 2], "Mark": [85, 89]})
    df

       ByCol  Mark Name
    0      1    85  AAA
    1      2    89  BBB

    df["Mark"].hist(by=df["ByCol"], bins=bins)
    plt.show()

but same code with a one row DataFrame such as

    df = pd.DataFrame({"Name": ["AAA"], "ByCol": [1], "Mark": [85]})
    df

       ByCol  Mark Name
    0      1    85  AAA

raises:

    ---------------------------------------------------------------------------
    AttributeError                            Traceback (most recent call last)
    <ipython-input-35-93bcd47d24f2> in <module>()
    ----> 1 df["Mark"].hist(by=df["Mark"], bins=bins)

    //anaconda/lib/python2.7/site-packages/pandas/tools/plotting.pyc in hist_series(self, by, ax, grid, xlabelsize, xrot, ylabelsize, yrot, figsize, bins, **kwds)
       2847                             **kwds)
       2848
    -> 2849     if axes.ndim == 1 and len(axes) == 1:
       2850         return axes[0]
       2851     return axes

    AttributeError: 'AxesSubplot' object has no attribute 'ndim'

Kind regards

### PR DESCRIPTION
**Pull request by s-celles**
Should closes https://github.com/pydata/pandas/issues/10214

### CODE REVIEW

File: pandas/io/excel.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from pandas.io.common import _is_url, _urlopen
 from pandas.tseries.period import Period
 from pandas import json
-from pandas.compat import map, zip, reduce, range, lrange, u, add_metaclass
+from pandas.compat import map, zip, reduce, range, lrange, u, add_metaclass, OrderedDict
```
Comment (jreback) [inline pandas/io/excel.py]: you need to remove all of this extraneous stuff, e.g. io/excel

**Comment by TomAugspurger** [comment]
We had a couple stray commits slip in. Will cleanup later.

**Comment by jreback** [comment]
@scls19fr can you rebase on master. ping when green.

**Comment by scls19fr** [comment]
Sorry @jreback I don't feel very confortable with Git.

I did:

    $ git rebase master

    $ git pull

    $ git push

What should I do ?

I would like to have my master branch sync with upstream.

**Comment by scls19fr** [comment]
Maybe this PR https://github.com/pydata/pandas/pull/10298 is better.
I deleted my repository. Fork again master. Create branch. Change 3 files and send PR.

**Pull request closed without merging**